avocado/plugins/diff.py

```diff
@@ -0,0 +1,427 @@
+        """
+        Add the subparser for the diff action.
+
+        :param parser: Main test runner parser.
+        """
+        parser = super(Diff, self).configure(parser)
+
+        parser.add_argument("jobids",
+                            default=[], nargs=2,
+                            metavar="<JOB>",
+                            help='Provide two Jobs, identified by their Job '
+                            '(partial) ID or test results directory.')
+
+        parser.add_argument('--html', type=str,
+                            dest='html_output',
+                            metavar='FILE',
+                            help='Enable HTML output to the FILE where the '
+                            'result should be written.')
+
+        parser.add_argument('--open-browser',
+                            action='store_true',
+                            default=False,
+                            help='Generate and open a HTML report in your '
+                            'preferred browser. If no --html file is '
+                            'provided, we create a temporary file. '
+                            'Current: False.')
+
+        parser.add_argument('--diff-filter',
+                            dest='diff_filter',
+                            type=self._validate_filters,
+                            default=['all'],
+                            help='Diff filter: (no)cmdline,(no)time,'
+                            '(no)variants,(no)results,(no)config,(no)sysinfo '
+                            '(defaults to all enabled).')
+
+        parser.add_argument('--paginator',
+                            choices=('on', 'off'), default='on',
+                            help='Turn the paginator on/off. '
+                            'Current: %(default)s')
+
+        parser.add_argument('--create-reports', action='store_true',
+                            help='Instead of show the diff using Avocado '
+                            'Diff utility, create temporary files to use '
+                            'in a custom diff tool.')
+
+        self.term = output.TERM_SUPPORT
+        self.std_diff_output = True
+
+    def run(self, args):
+        job1_dir, job1_id = self._setup_job(args.jobids[0])
+        job2_dir, job2_id = self._setup_job(args.jobids[1])
+
+        job1_data = self._get_job_data(job1_dir)
+        job2_data = self._get_job_data(job2_dir)
+
+        report_header = 'Avocado Job Report\n'
+        job1_results = [report_header]
+        job2_results = [report_header]
+
+        if (('all' in args.diff_filter or 'cmdline' in args.diff_filter) and
+           'nocmdline' not in args.diff_filter):
+
+            cmdline1 = self._get_command_line(job1_dir)
+            cmdline2 = self._get_command_line(job2_dir)
+
+            if str(cmdline1) != str(cmdline2):
+                command_line_header = ['\n',
+                                       'COMMAND LINE\n',
+                                       '============\n']
```

Comment:
This would be controversal, but the convention in RST is to put 1-2 empty lines before and 1 line after the heading. For section (=) heading it's usually the 2 before and 1 after and I'd prefer that.

Also at least my vim colores the headings, making it more visible. As we don't want to make "coloring books", how about to use the `log.info` (blue) color?

I mean first sections are pretty easily to distinguish, but the more sections we have the harder it gets to find them...